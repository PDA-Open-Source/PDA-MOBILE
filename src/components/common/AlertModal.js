import React, { PureComponent } from 'react';
import {
  Text,
  View,
  Modal,
  Dimensions,
  ViewPropTypes
} from 'react-native';
import PropTypes from 'prop-types';
import { verticalScale, TEXT_TYPE } from '../../theme/pdaStyleSheet';
import ButtonWithIcon from './ButtonWithIcon';
import { COLORS } from '../../constants/Color';
import CustomIcon from './CustomIcon';
const width = Dimensions.get('screen').width;

export default class AlertModal extends PureComponent{
  static propTypes = {
    icon : PropTypes.string,
    title : PropTypes.string,
    description : PropTypes.string,
    cancelButtonContainer : ViewPropTypes.style,
    onClosePress: PropTypes.func,
    onSuccessPress: PropTypes.func,
    onCancelPress: PropTypes.func,
    visible: PropTypes.bool,
    isIcon: PropTypes.bool,
    confirmName: PropTypes.string
  }

  static defaultProps = {
    onClosePress:()=>{},
    onSuccessPress:()=>{},
    onCancelPress:()=>{},
    icon:'trash-alt',
    title:'',
    description:'',
    visible: false,
    isIcon: true,
    confirmName: 'Delete'
  }


  renderModal(){
    const{
      description,
      onCancelPress,
      onSuccessPress,
      icon,
      isIcon,
      confirmName,
      title
    } = this.props;
    return(
      <View style={{margin: verticalScale(30), backgroundColor:COLORS.white,
        borderRadius: verticalScale(5), overflow:'hidden', width:(width/100)*90}}
      >
        <View style={{padding: 5}}>
          <Text style={[TEXT_TYPE.H5,{color:COLORS.black, padding:10}]}>{title}</Text>
          <View style={{paddingVertical: verticalScale(5), paddingLeft: verticalScale(20),flexDirection:'row'}}>
            {isIcon ? <View style={{paddingTop:verticalScale(13)}}>
              <CustomIcon
                size={verticalScale(25)}
                color={COLORS.red}
                name={icon}
                light
              />
            </View>:null}
            <View style={{paddingLeft:20}}>
              <Text style={[TEXT_TYPE.H3,{color:COLORS.black, paddingTop:10}]}>{description}</Text>
            </View>
          </View>
        </View>

        <View style={{
          flexDirection:'row',
          justifyContent:'flex-end',
          alignItems:'center',
          paddingHorizontal: verticalScale(25),
          paddingBottom: verticalScale(15)
        }}
        >
          <ButtonWithIcon
            textStyle={[TEXT_TYPE.H4,{color:COLORS.titleColor}]}
            name="Cancel"
            onPress={()=>onCancelPress()}
          />
          <ButtonWithIcon
            textStyle={[TEXT_TYPE.H4,{color:COLORS.red}]}
            name={confirmName}
            onPress={()=>onSuccessPress()}
          />

        </View>
      </View>
    );
  }

  render() {
    const {
      visible,
      onClosePress
    } = this.props;
    const modalProps = {
      visible
    };
    return (
      <Modal
        animationType="fade"
        supportedOrientations={['portrait']}
        onRequestClose={() => onClosePress()}
        transparent={true}
        {...modalProps}
      >
        <View
          style={{flex:1, alignItems:'center', justifyContent:'center',backgroundColor:COLORS.placeholderTextColor}}
        >
          {this.renderModal()}
        </View>
      </Modal>
    );
  }
}
