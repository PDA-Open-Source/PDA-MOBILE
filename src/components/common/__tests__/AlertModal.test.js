import React from 'react';
import renderer from 'react-test-renderer';
import AlertModal from '../AlertModal';
jest.mock('react-native-device-info');

describe('<AlertModal/>',()=>{
  it('AlertModal should render without crash',()=>{
    let snap = renderer.create(<AlertModal/>);
    expect(snap.toJSON()).toMatchSnapshot();
  });
});