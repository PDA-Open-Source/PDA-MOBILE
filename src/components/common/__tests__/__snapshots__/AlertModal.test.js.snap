// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "justifyContent": "center",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 100,
        "justifyContent": "center",
        "width": 100,
      }
    }
    supportedOrientations="portrait"
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "red",
          "flex": 1,
          "justifyContent": "center",
          "shadowColor": "red",
          "shadowOffset": Object {
            "height": 10,
            "width": 10,
          },
          "shadowOpacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "height": 200,
            "justifyContent": "center",
            "width": 280,
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        />
        <View>
          <Text />
        </View>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Close modal
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`<AlertModal/> AlertModal should render without crash 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
    ]
  }
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 49.04411764705882,
          "margin": 58.85294117647059,
          "overflow": "hidden",
          "width": 450,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EEEEEE",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 29.426470588235293,
            "paddingVertical": 9.808823529411764,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Regular",
                "fontSize": 26.714285714285715,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "color": "#121212",
              },
            ]
          }
        >
          
        </Text>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
            }
          }
        >
          <View>
            <View
              accessible={true}
              onLayout={[Function]}
              onPress={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onRippleAnimation={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="box-only"
              rippleCentered={false}
              rippleColor="rgb(0, 0, 0)"
              rippleDuration={400}
              rippleFades={true}
              rippleOpacity={0.3}
              rippleSequential={false}
              rippleSize={0}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "margin": 0,
                  "marginVertical": 5,
                  "paddingHorizontal": 10,
                  "paddingVertical": 2,
                }
              }
            >
              <View
                style={Object {}}
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#888888",
                        "fontSize": 35.311764705882354,
                      },
                      Object {},
                      Object {
                        "fontFamily": "Font Awesome 5 Pro",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {
                        "fontWeight": "100",
                      },
                    ]
                  }
                >
                  ÔÄç
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "bottom": 0,
                      "left": 0,
                      "overflow": "hidden",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "borderRadius": 78.47058823529412,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 19.61764705882353,
            "paddingVertical": 29.426470588235293,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Regular",
                "fontSize": 22,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "textAlign": "justify",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 49.04411764705882,
            "paddingVertical": 9.808823529411764,
          }
        }
      >
        <View>
          <View
            accessible={true}
            onLayout={[Function]}
            onPress={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="rgb(0, 0, 0)"
            rippleDuration={400}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={0}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#EEEEEE",
                "borderRadius": 58.85294117647059,
                "flexDirection": "row",
                "margin": 0,
                "marginHorizontal": 9.808823529411764,
                "marginVertical": 5,
                "paddingHorizontal": 39.23529411764706,
                "paddingVertical": 2,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "padding": 5,
                    "textAlign": "center",
                  },
                  Object {
                    "padding": 5,
                    "textAlign": "center",
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 22,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              Yes
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 78.47058823529412,
                  },
                ]
              }
            />
          </View>
        </View>
        <View>
          <View
            accessible={true}
            onLayout={[Function]}
            onPress={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="rgb(0, 0, 0)"
            rippleDuration={400}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={0}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#009688",
                "borderRadius": 58.85294117647059,
                "flexDirection": "row",
                "margin": 0,
                "marginHorizontal": 19.61764705882353,
                "marginVertical": 5,
                "paddingHorizontal": 49.04411764705882,
                "paddingVertical": 2,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "padding": 5,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 22,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              No
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 78.47058823529412,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`<AlertModal/> Button should render without crash 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
    ]
  }
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 49.04411764705882,
          "margin": 58.85294117647059,
          "overflow": "hidden",
          "width": 450,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EEEEEE",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 29.426470588235293,
            "paddingVertical": 9.808823529411764,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Regular",
                "fontSize": 26.714285714285715,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "color": "#121212",
              },
            ]
          }
        >
          
        </Text>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
            }
          }
        >
          <View>
            <View
              accessible={true}
              onLayout={[Function]}
              onPress={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onRippleAnimation={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="box-only"
              rippleCentered={false}
              rippleColor="rgb(0, 0, 0)"
              rippleDuration={400}
              rippleFades={true}
              rippleOpacity={0.3}
              rippleSequential={false}
              rippleSize={0}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "margin": 0,
                  "marginVertical": 5,
                  "paddingHorizontal": 10,
                  "paddingVertical": 2,
                }
              }
            >
              <View
                style={Object {}}
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#888888",
                        "fontSize": 35.311764705882354,
                      },
                      Object {},
                      Object {
                        "fontFamily": "Font Awesome 5 Pro",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {
                        "fontWeight": "100",
                      },
                    ]
                  }
                >
                  ÔÄç
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "bottom": 0,
                      "left": 0,
                      "overflow": "hidden",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "borderRadius": 78.47058823529412,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 19.61764705882353,
            "paddingVertical": 29.426470588235293,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Regular",
                "fontSize": 22,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "textAlign": "justify",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 49.04411764705882,
            "paddingVertical": 9.808823529411764,
          }
        }
      >
        <View>
          <View
            accessible={true}
            onLayout={[Function]}
            onPress={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="rgb(0, 0, 0)"
            rippleDuration={400}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={0}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#EEEEEE",
                "borderRadius": 58.85294117647059,
                "flexDirection": "row",
                "margin": 0,
                "marginHorizontal": 9.808823529411764,
                "marginVertical": 5,
                "paddingHorizontal": 39.23529411764706,
                "paddingVertical": 2,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "padding": 5,
                    "textAlign": "center",
                  },
                  Object {
                    "padding": 5,
                    "textAlign": "center",
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 22,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              Yes
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 78.47058823529412,
                  },
                ]
              }
            />
          </View>
        </View>
        <View>
          <View
            accessible={true}
            onLayout={[Function]}
            onPress={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="rgb(0, 0, 0)"
            rippleDuration={400}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={0}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#009688",
                "borderRadius": 58.85294117647059,
                "flexDirection": "row",
                "margin": 0,
                "marginHorizontal": 19.61764705882353,
                "marginVertical": 5,
                "paddingHorizontal": 49.04411764705882,
                "paddingVertical": 2,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "padding": 5,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 22,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              No
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 78.47058823529412,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
